examples/example-add-vectors.cpp

```diff
@@ -0,0 +1,164 @@
+//
+// * Redistributions in binary form must reproduce the above copyright notice,
+//   this list of conditions and the disclaimer (as noted below) in the
+//   documentation and/or other materials provided with the distribution.
+//
+// * Neither the name of the LLNS/LLNL nor the names of its contributors may
+//   be used to endorse or promote products derived from this software without
+//   specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL LAWRENCE LIVERMORE NATIONAL SECURITY,
+// LLC, THE U.S. DEPARTMENT OF ENERGY OR CONTRIBUTORS BE LIABLE FOR ANY
+// DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+// DAMAGES  (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS
+// OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)
+// HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+// STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING
+// IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+// POSSIBILITY OF SUCH DAMAGE.
+//
+//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~//
+
+#include <cstdlib>
+#include <iostream>
+
+#include "memoryManager.hpp"
+
+#include "RAJA/RAJA.hpp"
+#include "RAJA/util/defines.hpp"
+
+const int CUDA_BLOCK_SIZE = 256;
+
+void checkSolution(int *C, int in_N);
+
+/*
+  Example 1: Adding Two Vectors
+
+  ----[Details]---------------------
+  Adds two vectors of length N
+
+  -----[RAJA Concepts]---------------
+  1. Introduces the forall loop and basic RAJA policies
+
+  RAJA::forall<RAJA::exec_policy>(RAJA::Range, [=] (RAJA::Index_type i)) {
+
+         //body
+
+  });
+
+  [=] Pass by copy
+  [&] Pass by reference
+  RAJA::exec_policy - specifies how the traversal occurs
+  RAJA::Range       - Provides a list of iterables
+  RAJA::Index_type  - Index for RAJA loops
+ */
+int main(int RAJA_UNUSED_ARG(argc), char **RAJA_UNUSED_ARG(argv[]))
+{
+
+  printf("Example 1: Adding Two Vectors \n \n");
+
+  const int N = 1000;
+  int *A = memoryManager::allocate<int>(N);
+  int *B = memoryManager::allocate<int>(N);
+  int *C = memoryManager::allocate<int>(N);
+
+  for (int i = 0; i < N; ++i) {
+    A[i] = -i;
+    B[i] =  i;
+  }
+
+  printf("Standard C++ Loop \n");
+  for (int i = 0; i < N; ++i) {
+    C[i] = A[i] + B[i];
+  }
+  checkSolution(C, N);
+
+
+  printf("RAJA: Sequential Policy \n");
+  /*
+    RAJA::seq_exec -  Executes the loop sequentially
+
+    RAJA::RangeSegment(start,stop) - Generates a contiguous sequence of numbers
+    by the [start, stop) interval specified
+  */
+  RAJA::forall<RAJA::seq_exec>
+    (RAJA::RangeSegment(0, N), [=](RAJA::Index_type i) {
+      C[i] = A[i] + B[i]; 
```

Comment:
It is more appropriate to have the parenthesis on the line above than on the line below.
```cpp
RAJA::forall<RAJA::seq_exec>(
  RAJA::RangeSegment(0, N), [=] (RAJA::Index_type i) {
    C[i] = A[i] + B[i];
  });
```

is arguably cleaner. This is also closer to what autoformatting tools, such as `clang-format`, will yield